src/Dialect/ONNX/ONNXOps/Math/Reduction.cpp

```diff
@@ -393,6 +393,31 @@ LogicalResult ONNXReduceSumSquareV13Op::inferShapes(
   return inferShapeForReductionOps_old<ONNXReduceSumSquareV13Op>(*this);
 }
 
+//===----------------------------------------------------------------------===//
+// Folder
+//===----------------------------------------------------------------------===//
+
+OpFoldResult ONNXReduceMeanOp::fold(FoldAdaptor adaptor) {
+  typename ONNXReduceMeanOp::Adaptor opAdaptor(*this);
+  onnx_mlir::ONNXGenericReductionOpShapeHelper<ONNXReduceMeanOp> shapeHelper(
+      getOperation(), opAdaptor.getOperands());
+
+  if (failed(shapeHelper.computeShape()))
+    return nullptr;
+
+  size_t nbAxesToReduce = 0;
+  for (bool red : shapeHelper.isReductionAxis) {
```

Comment:
I wonder if you could use `llvm::count_if`:
```suggestion
  nbAxesToReduce = llvm::count_if(shapeHelper.isReductionAxis, [&](auto reductionAxis)) {
  return reductionAxis;
  });
```

Reply:
Continuing on what Tiago suggested, I'd say use a bool (like hasReduction) instead of int and use llvm::any_of instead of llvm::count_if

Reply:
that's a good idea! thanks!